tests/Unit/Evolution/Systems/GeneralizedHarmonic/Actions/Test_NumericInitialData.cpp

```diff
@@ -0,0 +1,244 @@
+                                  GeneralizedHarmonic::all_numeric_vars>,
+      Metavariables>(make_not_null(&runner), element_id)[0];
+  GeneralizedHarmonic::Solutions::WrappedGr<gr::Solutions::KerrSchild> kerr{
+      1., {{0., 0., 0.}}, {{0., 0., 0.}}};
+  const auto kerr_gh_vars = kerr.variables(coords, 0., gh_system_vars{});
+  if (selected_vars == NumericInitialDataVariables::GeneralizedHarmonic) {
+    get<gr::Tags::SpacetimeMetric<3, Frame::Inertial, DataVector>>(inbox) =
+        get<gr::Tags::SpacetimeMetric<3, Frame::Inertial, DataVector>>(
+            kerr_gh_vars);
+    get<Tags::Pi<3, Frame::Inertial>>(inbox) =
+        get<Tags::Pi<3, Frame::Inertial>>(kerr_gh_vars);
+    get<Tags::Phi<3, Frame::Inertial>>(inbox) =
+        get<Tags::Phi<3, Frame::Inertial>>(kerr_gh_vars);
+  } else if (selected_vars == NumericInitialDataVariables::Adm) {
+    const auto kerr_adm_vars = kerr.variables(
+        coords, 0.,
+        tmpl::list<
+            gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>,
+            gr::Tags::Lapse<DataVector>,
+            gr::Tags::Shift<3, Frame::Inertial, DataVector>,
+            gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DataVector>>{});
+    get<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>>(inbox) =
+        get<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>>(
+            kerr_adm_vars);
+    get<gr::Tags::Lapse<DataVector>>(inbox) =
+        get<gr::Tags::Lapse<DataVector>>(kerr_adm_vars);
+    get<gr::Tags::Shift<3, Frame::Inertial, DataVector>>(inbox) =
+        get<gr::Tags::Shift<3, Frame::Inertial, DataVector>>(kerr_adm_vars);
+    get<gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DataVector>>(inbox) =
+        get<gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DataVector>>(
+            kerr_adm_vars);
+  } else {
+    REQUIRE(false);
+  }
+
+  // SetNumericInitialData
+  ActionTesting::next_action<element_array>(make_not_null(&runner), element_id);
+
+  // Check result
+  Approx custom_approx = Approx::custom().epsilon(1.e-3).scale(1.0);
+  CHECK_ITERABLE_CUSTOM_APPROX(
+      get_element_tag(
+          gr::Tags::SpacetimeMetric<3, Frame::Inertial, DataVector>{}),
+      (get<gr::Tags::SpacetimeMetric<3, Frame::Inertial, DataVector>>(
+          kerr_gh_vars)),
+      custom_approx);
+  CHECK_ITERABLE_CUSTOM_APPROX(
+      get_element_tag(Tags::Pi<3, Frame::Inertial>{}),
+      (get<Tags::Pi<3, Frame::Inertial>>(kerr_gh_vars)), custom_approx);
+  CHECK_ITERABLE_CUSTOM_APPROX(
+      get_element_tag(Tags::Phi<3, Frame::Inertial>{}),
+      (get<Tags::Phi<3, Frame::Inertial>>(kerr_gh_vars)), custom_approx);
+}
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.Gh.NumericInitialData",
+                  "[Unit][Evolution]") {
+  test_numeric_initial_data(
+      GeneralizedHarmonic::NumericInitialDataVariables::GeneralizedHarmonic);
+  test_numeric_initial_data(
+      GeneralizedHarmonic::NumericInitialDataVariables::Adm);
+}
+
+}  // namespace GeneralizedHarmonic
```

Comment:
Arbitrary line: Add an error test where you fail to create a `GeneralizedHarmonic::NumericInitialDataVariables` from options because of a bad input name like:
```
Variables: KyleAlwaysMisspellsThngs
```

Reply:
I believe this is already sufficiently tested as part of the option parsing, don't you agree?

Reply:
Yes it is.